### PR TITLE
Shipment/Waypoint Controller, AirportSeeder, TrackShipment-View imple…

### PR DESCRIPTION
**Pull request by inancalp**
…mentations

(!)ShipmentController :
-----------------------
- track() function that retrieves waypoints and converts them into geocodes to use within shipments.track route.

(!) shipments\track-shipment.blade.php:
--------------------------------------------
- User will be see an overview of the state for the shipment.
- Different colored polylines will be presented within BingMaps Map for different waypoint states.

WaypointController :
-----------------------
- updates on update() and store() methods.

(!) GuzzleHttp in use for API requests.

(!) config/country_codes.php file.
------------------------------------
- for to retrieve the ISO alpha-2 country codes for AirLabs API.

(!) AirportSeeder
------------------
- Airlabs API retrieves the geocodes for the countries, which are retrieved form DepotSeeder.
- BingMaps API converts geocodes to actual addresses.
- Overall idea: Create Depots, retrieve Airports within the same country.

(!) shipments\set.blade.php :
-------------------------------
- Only the depots and airports within the shipment's source and destination addresses will be shown to the employee.

### CODE REVIEW

**Comment by ThomasFrans** [comment]
Does this PR replace #148?

**Comment by inancalp** [comment]
> Does this PR replace #148?
Yes

File: app/Http/Controllers/ShipmentController.php [hunk 1]
```diff
@@ -336,40 +336,102 @@ public function sendInvoiceMail(Invoice $invoice): View|RedirectResponse
 
     // Bing Maps Locations API
     // Template API that CONVERTS ADDRESS TO GEOCODE(latitude, longitude) to be able to display each waypoint relevant to the shipment in concern.
-    public function track(): View
+    public function track(Shipment $shipment)
     {
+        $waypoints = $shipment->waypoints;
+        // dd($waypoints);
+        $waypoints_geocodes = collect([]);
+        // dd($waypoints[0]->current_address_id);
```
Comment (ThomasFrans) [inline app/Http/Controllers/ShipmentController.php:344]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: app/Http/Controllers/ShipmentController.php [hunk 2]
```diff
@@ -336,40 +336,102 @@ public function sendInvoiceMail(Invoice $invoice): View|RedirectResponse
 
     // Bing Maps Locations API
     // Template API that CONVERTS ADDRESS TO GEOCODE(latitude, longitude) to be able to display each waypoint relevant to the shipment in concern.
-    public function track(): View
+    public function track(Shipment $shipment)
     {
+        $waypoints = $shipment->waypoints;
+        // dd($waypoints);
```
Comment (ThomasFrans) [inline app/Http/Controllers/ShipmentController.php:342]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: app/Http/Controllers/ShipmentController.php [hunk 3]
```diff
@@ -336,40 +336,102 @@ public function sendInvoiceMail(Invoice $invoice): View|RedirectResponse
...
+            $locality = str_ireplace(' ', '%20', $current_locality);
+
+            $current_postalCode = str_replace('.', '', trim($current_address->postal_code));
+            $postalCode = str_ireplace(' ', '%20', $current_postalCode);
+
+            //request URL is created here + response is retrieved with the DATA
+            $findURL = $baseURL.'/'.$country.'/'.$state.'/'.$postalCode.'/'.$locality.'/'
+            .$street.'?output=xml&key='.$key;
+
+            // dump($findURL);
```
Comment (ThomasFrans) [inline app/Http/Controllers/ShipmentController.php:375]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: app/Http/Controllers/ShipmentController.php [hunk 4]
```diff
@@ -336,40 +336,102 @@ public function sendInvoiceMail(Invoice $invoice): View|RedirectResponse
...
+            $output = file_get_contents($findURL);
+            $response = new \SimpleXMLElement($output);
+
+            $latitude = $response->ResourceSets->ResourceSet->Resources->Location->Point->Latitude->__toString();
+            $longitude = $response->ResourceSets->ResourceSet->Resources->Location->Point->Longitude->__toString();
+
+            // dd($latitude);
```
Comment (ThomasFrans) [inline app/Http/Controllers/ShipmentController.php:382]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: app/Http/Controllers/ShipmentController.php [hunk 5]
```diff
@@ -336,40 +336,102 @@ public function sendInvoiceMail(Invoice $invoice): View|RedirectResponse
...
+        ];
 
-        // here is the implementation to reverse geocodes into address again.
-        // for debugging purposes.
-        $centerPoint = $latitude.','.$longitude;
-        $revGeocodeURL = $baseURL.'/'.$centerPoint.'?output=xml&key='.$key;
-        $rgOutput = file_get_contents($revGeocodeURL);
-        $rgResponse = new \SimpleXMLElement($rgOutput);
-        $address = $rgResponse->ResourceSets->ResourceSet->Resources->Location->Address->FormattedAddress;
+        // dd($waypoints_geocodes);
```
Comment (ThomasFrans) [inline app/Http/Controllers/ShipmentController.php:432]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

File: app/Http/Controllers/ShipmentController.php [hunk 6]
```diff
@@ -336,40 +336,102 @@ public function sendInvoiceMail(Invoice $invoice): View|RedirectResponse
...
+        $locality = str_ireplace(' ', '%20', $destination_locality);
+
+        $destination_postalCode = str_replace('.', '', $destination_address->postal_code);
+        $postalCode = str_ireplace(' ', '%20', $destination_postalCode);
 
         //request URL is created here + response is retrieved with the DATA
         $findURL = $baseURL.'/'.$country.'/'.$state.'/'.$postalCode.'/'.$locality.'/'
         .$street.'?output=xml&key='.$key;
+
+        // dump($findURL);
```
Comment (ThomasFrans) [inline app/Http/Controllers/ShipmentController.php:415]: https://github.com/Practice-Enterprise2/AD#pull-request-requirements

**Review by ThomasFrans — CHANGES_REQUESTED** [review]

File: routes/web.php [hunk 7]
```diff
@@ -145,6 +145,8 @@
 
     Route::controller(ShipmentController::class)->group(function () {
         Route::get('/shipments', 'index')->name('shipments.index')->can('viewAny', Shipment::class);
+        Route::get('/shipments/requests', 'requests')->name('shipments.requests')->can('acceptAny', Shipment::class);
+        Route::post('shipments/requests/{shipment}/evaluate', 'evaluate')->name('shipments.requests.evaluate')->can('accept', 'shipment');
```
Comment (ThomasFrans) [inline routes/web.php:149]: Duplicates (see some lines below).

**Review by ThomasFrans — CHANGES_REQUESTED** [review]

**Review by ThomasFrans — APPROVED** [review]

**Pull request merged**